src/aoemotebutton.cpp

```diff
@@ -26,50 +24,50 @@ void AOEmoteButton::set_image(QString p_image, QString p_emote_comment)
         "QToolTip { color: #000000; background-color: #ffffff; border: 0px; }");
   }
   else {
-    if(p_image.contains("_off") || p_image.contains("_on")){
-        //Since we can't detect a button, lets just generate one!
-        QString tmp_p_image = p_image;
-        //if neither buttons exist then just throw some text
-        if(!file_exists(tmp_p_image.replace("_off","_on")) && !file_exists(tmp_p_image.replace("_on","_off")))
-        {
-            this->setText(p_emote_comment);
-            this->setStyleSheet("QPushButton { border-image: url(); }"
-                                "QToolTip { background-image: url(); color: #000000; "
-                                "background-color: #ffffff; border: 0px; }");
-            return;
-        }
-
-        QImage *tmpImage = new QImage(tmp_p_image);
-            QPoint p1, p2;
-            p2.setY(tmpImage->height());
-
-            QLinearGradient gradient(p1, p2);
-            gradient.setColorAt(0, Qt::transparent);
-            gradient.setColorAt(1, QColor(0, 0, 0, 159));
-
-            QPainter p(tmpImage);
-            p.fillRect(0, 0, tmpImage->width(), tmpImage->height(), gradient);
-
-            gradient.setColorAt(0,QColor(0, 0, 0, 159));
-            gradient.setColorAt(1, Qt::transparent);
-            p.fillRect(0,0, tmpImage->width(), tmpImage->height(), gradient);
-
-            p.end();
-       tmpImage->save(p_image,"png");
-       set_image(p_image,p_emote_comment);
-
-    } else {
+    if (p_image.contains("_off") || p_image.contains("_on")) {
+      // Since we can't detect a button, lets just generate one!
+      QString tmp_p_image = p_image;
+      // if neither buttons exist then just throw some text
+      if (!file_exists(tmp_p_image.replace("_off", "_on")) &&
+          !file_exists(tmp_p_image.replace("_on", "_off"))) {
         this->setText(p_emote_comment);
-        this->setStyleSheet("QPushButton { border-image: url(); }"
-                            "QToolTip { background-image: url(); color: #000000; "
-                            "background-color: #ffffff; border: 0px; }");
+        this->setStyleSheet(
+            "QPushButton { border-image: url(); }"
+            "QToolTip { background-image: url(); color: #000000; "
+            "background-color: #ffffff; border: 0px; }");
+        return;
+      }
+
+      QImage tmpImage = QImage(tmp_p_image);
```

Comment:
more succinctly (just a remark): `QImage tmpImage(tmp_p_image);`

Otherwise this sounds like you are doing a copy initialization which is not necessary.